tests/check/check_h26xsigned_auth.c

```diff
@@ -1625,6 +1625,236 @@ START_TEST(vendor_axis_communications_operation)
+      ck_assert(latest);
+
+      if (wrong_key) {
+        ck_assert(latest->public_key_has_changed);
+        ck_assert_int_eq(latest->authenticity, SV_AUTH_RESULT_NOT_OK);
+      } else {
+        ck_assert(!(latest->public_key_has_changed));
+        ck_assert_int_eq(latest->authenticity, SV_AUTH_RESULT_OK);
+      }
+    }
+    // We are done with auth_report
+    signed_video_authenticity_report_free(auth_report);
+
+    // Free nalu_list_item and session.
+    signed_video_free(sv);
+    nalu_list_free_item(sei);
+    nalu_list_free_item(i_nalu);
+  }
+}
+
+/* Test description 
+ * Check if the API signed_video_add_public_key_to_sei can add a public key to the SEI and if
+ * signed_video_add_public_key_to_sei can set a public key on the auth side.
+ *
+ * Verify that it is not valid to add a manipulated key or set a public key in the middle of a
+ * stream.
+ */
+START_TEST(set_public_key)
+{
+  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
+  // |settings|; See signed_video_helpers.h.
+
+  struct pk_setting {
+    bool pk_in_sei;
+    bool wrong_pk;
+    bool use_signature_info;
+    bool set_pk;
+  };
+
+  struct pk_setting pk_tests[] = {
+    // Public key on validation side from start
+    {false, false, false, true},
+    // Public key in sei and on validation side from start
+    {true, false, false, true},
+    // No public key
+    {false, false, false, false},
+    // Public key on validation side later
+    {false, false, true, false},
+    // Public key in sei and on validation side later
+    {true, false, true, false},
+    // Public key in sei and bad public key on validation side
+    {true, true, true, false},
+  };
+
+  for (int j = 0; j < (int) (sizeof(pk_tests) / sizeof(*pk_tests)); j++) {
+    SignedVideoReturnCode sv_rc;
+    SignedVideoCodec codec = settings[_i].codec;
+    nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
+    nalu_list_item_t *sei = NULL;
+    signed_video_t *sv_camera = NULL;
+    sign_algo_t algo = settings[_i].algo;
+    char *private_key = NULL;
+
+    sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], pk_tests[j].pk_in_sei,
+        i_nalu, &sei);
+
+    // On validation side
+    signed_video_t *sv_vms = signed_video_create(codec);
+
+    signature_info_t sign_info_wrong_key = {0};
+    if (pk_tests[j].wrong_pk) {
+      // Generate a new private key in order to extract a bad private key (a key not compatible with
+      // the one generated on the camera side)
+      signed_video_generate_private_key( algo, "./", (char **)&sign_info_wrong_key.private_key,
+          &sign_info_wrong_key.private_key_size);
+      openssl_read_pubkey_from_private_key(&sign_info_wrong_key);
+      // Set public key
+      sv_rc = signed_video_set_public_key(sv_vms, sign_info_wrong_key.public_key,
+          sign_info_wrong_key.public_key_size); ck_assert_int_eq(sv_rc, SV_OK);
+    }
+
+    if (pk_tests[j].set_pk) {
```

Comment:
Change order of the blocks, that is determine key type first (correct vs wrong) and then set it.
```
signature_info_t *sign_info = sv_camera->signature_info;
if (pk_tests[j].use_wrong_pk) {
  sign_info = &sign_info_wrong_key;
}
if (pk_tests[j].set_pk_before_session_start) {
  // Set public key
  sv_rc = signed_video_set_public_key( sv_vms, sign_info->public_key, sign_info->public_key_size); 
  ck_assert_int_eq(sv_rc, SV_OK);
}
if (!pk_tests[j].set_pk_after_session_start) {
  sign_info = NULL;
}
```